components/range_cache_memory_engine/src/write_batch.rs

```diff
@@ -954,4 +954,43 @@ mod tests {
             .unwrap();
         assert_eq!(snap2.get_value(b"zkk11").unwrap().unwrap(), &val1);
     }
+
+    #[test]
+    fn test_write_batch_update_outdated_pending_region() {
```

Comment:
The current master branch also pass the test.

Reply:
Done, add a check for `regions_by_range`